core/src/main/scala/org/bitcoins/core/protocol/ln/LnCurrencyUnit.scala

```diff
@@ -0,0 +1,239 @@
+package org.bitcoins.core.protocol.ln
+
+import org.bitcoins.core.consensus.Consensus
+import org.bitcoins.core.currency.{ Bitcoins, Satoshis }
+import org.bitcoins.core.number.{ BaseNumbers, Int64 }
+import org.bitcoins.core.protocol.NetworkElement
+
+sealed abstract class LnCurrencyUnit extends NetworkElement {
+  type A
+
+  def character: Char
+
+  def multiplier: BigDecimal
+
+  def >=(ln: LnCurrencyUnit): Boolean = {
+    toPicoBitcoins >= ln.toPicoBitcoins
+  }
+
+  def >(ln: LnCurrencyUnit): Boolean = {
+    toPicoBitcoins > ln.toPicoBitcoins
+  }
+
+  def <(ln: LnCurrencyUnit): Boolean = {
+    toPicoBitcoins < ln.toPicoBitcoins
+  }
+
+  def <=(ln: LnCurrencyUnit): Boolean = {
+    toPicoBitcoins <= ln.toPicoBitcoins
+  }
+
+  def !=(ln: LnCurrencyUnit): Boolean = !(this == ln)
+
+  def ==(ln: LnCurrencyUnit): Boolean = toPicoBitcoins == ln.toPicoBitcoins
+
+  def +(ln: LnCurrencyUnit): LnCurrencyUnit = {
+    val returnType = if (ln.multiplier < multiplier) ln else this //return the type with the most precision.
+    currencyUnitFromPicoValue(returnType, toPicoBitcoins + ln.toPicoBitcoins)
+  }
+
+  def -(ln: LnCurrencyUnit): LnCurrencyUnit = {
+    val returnType = if (ln.multiplier < multiplier) ln else this //return the type with the most precision.
+    currencyUnitFromPicoValue(returnType, toPicoBitcoins - ln.toPicoBitcoins)
+  }
+
+  def *(ln: LnCurrencyUnit): LnCurrencyUnit = {
+    PicoBitcoins(toPicoBitcoins * ln.toPicoBitcoins)
+  }
+
+  def unary_- : LnCurrencyUnit = {
+    currencyUnitFromPicoValue(this, -toPicoBitcoins)
+  }
+
+  override def bytes: Seq[Byte] = toSatoshis.bytes
+
+  def toBigInt: BigInt
+
+  protected def underlying: A
+
+  def toSatoshis: Satoshis
+
+  def toPicoBitcoins: BigInt
+
+  def currencyUnitFromPicoValue(lnType: LnCurrencyUnit, unit: BigInt): LnCurrencyUnit = {
```

Comment:
I'm going to see if Nadav knows of a better way to do this, but at minimum I think we should make this `private` so no one else can call this

Reply:
Let's do that for v1 of this